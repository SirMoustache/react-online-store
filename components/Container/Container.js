/**
 * Absolute imports
 */
import styled from 'styled-components';

/**
 * Theme
 */
import theme from '../../theme';

const Container = styled.div`
  margin-left: auto;
  margin-right: auto;
  max-width: ${theme.sizes.container.sm};

  @media (min-width: 1200px) {
    max-width: ${theme.sizes.container.lg};
  }
`;

export default Container;
