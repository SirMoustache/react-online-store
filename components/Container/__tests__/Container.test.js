/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';

/**
 * Local Components
 */
import Container from '..';

const child = <span className="unique">Test</span>;

const renderComponent = ({ ...rest } = {}) =>
  render(<Container {...rest}>{child}</Container>);

describe('<Container />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a <div> tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('div')).not.toBeNull();
  });

  it('should have a className attribute', () => {
    const className = 'test';
    const { container } = renderComponent({ className });
    expect(container.querySelector('div').classList).toContain(className);
  });

  it('should render children when passed in', () => {
    const { container } = renderComponent();
    expect(container.querySelector('.unique')).not.toBeNull();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const { container } = renderComponent({ id });
    expect(container.querySelector('div').id).toContain(id);
  });

  it('should not adopt an invalid attribute', () => {
    const attribute = 'test';
    const { container } = renderComponent({ attribute });
    expect(container.querySelector('div').getAttribute('attribute')).toBeNull();
  });
});
