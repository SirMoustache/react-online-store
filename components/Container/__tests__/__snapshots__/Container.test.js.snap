// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div>
  <div
    class="Container-sc-1kdd3x2-0 kwPFyY"
  >
    <span
      class="unique"
    >
      Test
    </span>
  </div>
</div>
`;
